contracts/Tokens/Prime/README.md

```diff
@@ -0,0 +1,334 @@
+
+The graph above demonstrates the relationship between an increased XVS staked amount and its effect on market rewards, assuming a constant participation of $2.5K USD in the BTC supply market. This helps visualize how an increase in the staked amount influences the APY.
+
+### Implementation of the rewards in solidity
+
+There is a global `rewardIndex` and `sumOfMembersScore` variables per supported market. `sumOfMembersScore` represents the current sum of all the prime token holders score. And `rewardIndex` needs to be updated whenever a user’s staked XVS or supply/borrow changes. 
+
+```jsx
+totalIncomeToDistribute = N
+
+// every time accrueInterest is called. delta is interest per score
+delta = totalIncomeToDistribute / sumOfMembersScore
+rewardIndex += delta
+```
+
+Whenever a user’s supply/borrow or XVS vault balance changes we will calculate the rewards accrued and add them to their account. This is how we will calculate:
+
+```jsx
+rewards = (rewardIndex - userRewardIndex) * scoreOfUser
+```
+
+Then we will update the `userRewardIndex` (`interests[market][account]`) to their current global values.
+
+## Income collection and distribution
+
+Interest reserves (part of the protocol income) from core pool markets is sent to PSR ([Protocol Share Reserve](https://github.com/VenusProtocol/protocol-reserve)). Based on configuration a certain percentage of spread income from Prime markets is reserved for prime token holders. The interest reserves will be sent to the PSR every 10 blocks (this can be changed by the community via [VIP](https://app.venus.io/governance)).
+
+The PSR has a function `releaseFunds` that needs to be invoked to release the funds to the Prime contract and it also has a function `getUnreleasedFunds` to get the unreleased funds for a given destination target.
+
+Prime contract takes the total released + unreleased funds and distributes to prime token holders each block. The distribution is proportional to the score of the prime token holders.
+
+When a user claims their rewards and if the contract doesn’t have enough funds then we trigger the release of funds from PSR to Prime contract in the same transaction i.e., in the `claim` function.
+
+More information about Income collection and distribution [here](https://docs-v4.venus.io/whats-new/automatic-income-allocation).
+
+## Update cap multipliers and alpha
+
+We need the ability to update market multipliers and alpha anytime. When these values are updated changes will be gradually applied instead of applying changes instantly. So as users will borrow/supply their individual scores and the sum of total scores will be updated.
+
+There will be also a script that will call the permission-less function `updateScores` to update the scores of all users at any time. This script won’t pause the market or prime contract. This script updates scores in multiple transactions as in 1 transaction it will run out of gas. 
```

Comment:
```suggestion
To mitigate this issue, Venus will supply a script that will use the permission-less function `updateScores` to update the scores of all users. This script won’t pause the market or prime contract. This script will update scores in multiple transactions because it will run out of gas trying to update all scores in 1 transaction. 
```